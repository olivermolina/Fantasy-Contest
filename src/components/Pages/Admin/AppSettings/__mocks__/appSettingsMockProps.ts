export const appSettingsMockProps = {
  appSettings: {
    BONUS_CREDIT_FREE_ENTRY_EQUIVALENT: '25',
    STAKE_TYPE_FREE_ENTRY: 'ALL_IN,INSURED',
    NUMBER_OF_PLAYERS_FREE_ENTRY: '2,3,4',
    MAX_MATCH_DEPOSIT_AMOUNT: '100',
    DEPOSIT_AMOUNT_OPTIONS: '10,25,50,75,100,200',
    RELOAD_BONUS_TYPE: 'FLAT',
    RELOAD_BONUS_AMOUNT: '0',
    MAX_RETENTION_BONUS: '50',
    RETENTION_BONUS_WEEKLY_CHANCE: '1',
    RETENTION_BONUS_MATCH_MULTIPLIER: '1',
    REFERRAL_CREDIT_AMOUNT: '25',
    REFERRAL_CUSTOM_TEXT:
      'Refer a friend from the link or code in your profile and get $25 in bonus credit!!',
    MIN_BET_AMOUNT: '1',
    MAX_BET_AMOUNT: '50',
    MAX_MARKET_ODDS: '150',
    MIN_MARKET_ODDS: '-150',
    initial: false,
    CHALLENGE_PROMO_MESSAGE:
      'Pick 2-4 players. Predict if they will get MORE or LESS than their projection.',
    REPEAT_ENTRIES_LIMIT: '1',
  },
  onSubmit: () => console.log('onSubmit'),
};
