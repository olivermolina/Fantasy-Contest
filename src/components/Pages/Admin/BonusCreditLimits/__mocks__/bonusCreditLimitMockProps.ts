import { BetStakeType } from '@prisma/client';

export const bonusCreditLimitMockProps = {
  onSubmit: () => console.log('submit'),
  data: {
    signupFreeEntry: false,
    bonusCreditFreeEntryEquivalent: 50,
    bonusCreditLimits: [
      {
        contestCategoryId: '1',
        numberOfPicks: 2,
        id: '30387acd-660c-46f0-a9e6-af87c007e6fb',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 25,
      },
      {
        contestCategoryId: '2',
        numberOfPicks: 3,
        id: '316555e5-b447-468a-b57f-381bbf64919a',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 50,
      },
      {
        contestCategoryId: '3',
        numberOfPicks: 4,
        id: 'fc063fcd-1a25-46e2-9361-d947eec1f9a9',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 40,
      },
      {
        contestCategoryId: '4',
        numberOfPicks: 5,
        id: 'a2542740-e90f-4763-8a3a-f7afba68898c',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 20,
      },
    ],
  },
  openAddUsersFreeEntryConfirmDialog: () => console.log('open'),
};
