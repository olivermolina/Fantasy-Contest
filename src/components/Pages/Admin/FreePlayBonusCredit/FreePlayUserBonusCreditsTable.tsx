import React, { useMemo, useState } from 'react';
import {
  DataGrid,
  GridColDef,
  GridRowsProp,
  GridSlotsComponentsProps,
} from '@mui/x-data-grid';
import { Button, Stack } from '@mui/material';
import CancelFreePlayConfirmDialog from '~/components/Pages/Admin/FreePlayBonusCredit/CancelFreePlayConfirmDialog';
import { TransactionType } from '@prisma/client';
import { PaymentStatusCode } from '~/constants/PaymentStatusCode';

declare module '@mui/x-data-grid' {
  interface FooterPropsOverrides {
    totalBonusCredits: number;
  }
}

type Props = {
  onSubmitDeletePlayFreeCredit(currentRow: FreePlayRowModel): unknown;
  data: readonly FreePlayRowModel[];
};

export interface FreePlayRowModel {
  id: string;
  transactionDate: Date;
  amountBonus: number;
  status: PaymentStatusCode;
  type: TransactionType;
}

function CustomFooter(props: NonNullable<GridSlotsComponentsProps['footer']>) {
  return (
    <div className={'flex p-2 font-semibold text-lg'}>
      Total Bonus Credits: {Number(props.totalBonusCredits).toFixed(2)}
    </div>
  );
}

export default function FreePlayUserBonusCreditsTable(props: Props) {
  const rows: GridRowsProp<FreePlayRowModel> = props.data;
  const [selectedRow, setSelectedRow] = useState<
    FreePlayRowModel | undefined
  >();
  const [open, setOpen] = useState(false);

  const columns: GridColDef[] = [
    { flex: 1, field: 'id', headerName: 'Transaction ID' },
    {
      flex: 1,
      field: 'transactionDate',
      headerName: 'Transaction Date',
      valueFormatter: (params) => params.value.toLocaleString(),
    },
    { flex: 1, field: 'type', headerName: 'Type' },
    { flex: 1, field: 'amountBonus', headerName: 'Amount' },
    {
      flex: 1,
      field: 'status',
      headerName: 'Status',
      renderCell: (params) => {
        const currentRow = params.row as typeof rows[0];
        return Object.keys(PaymentStatusCode)[
          Object.values(PaymentStatusCode).indexOf(currentRow.status as number)
        ];
      },
    },
    {
      flex: 1,
      field: 'action',
      headerName: 'Action',
      renderCell: (params) => {
        const currentRow = params.row as typeof rows[0];
        const onClick = () => {
          setSelectedRow(currentRow);
          setOpen(true);
        };

        return (
          <Stack direction="row" spacing={2}>
            <Button
              color="warning"
              size="small"
              onClick={onClick}
              variant={'contained'}
              disabled={currentRow.status !== PaymentStatusCode.COMPLETE}
            >
              Cancel
            </Button>
          </Stack>
        );
      },
    },
  ];

  const handleClose = () => {
    setOpen(false);
  };

  const deleteBonusCredit = (row: FreePlayRowModel) => {
    props.onSubmitDeletePlayFreeCredit(row);
    handleClose();
  };

  const totalBonusCredits = useMemo(
    () =>
      props.data.reduce(
        (acc, curr) =>
          curr.status === PaymentStatusCode.COMPLETE
            ? curr.type === TransactionType.CREDIT
              ? acc + curr.amountBonus
              : acc - curr.amountBonus
            : acc,
        0,
      ),
    [props.data],
  );

  return (
    <div className="flex flex-col gap-2">
      <span className={'text-lg font-bold'}>Bonus Credit Transactions</span>
      <div style={{ height: 300, width: '100%' }}>
        <DataGrid
          getRowId={(row) => row.id}
          rows={rows}
          columns={columns}
          slots={{
            footer: CustomFooter,
          }}
          slotProps={{
            footer: { totalBonusCredits },
          }}
        />
        <CancelFreePlayConfirmDialog
          open={open}
          setOpen={setOpen}
          handleDelete={deleteBonusCredit}
          row={selectedRow}
        />
      </div>
    </div>
  );
}
