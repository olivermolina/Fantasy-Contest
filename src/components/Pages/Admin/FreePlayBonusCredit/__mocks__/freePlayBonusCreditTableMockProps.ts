import { PaymentStatusCode } from '~/constants/PaymentStatusCode';
import { TransactionType } from '@prisma/client';

export const freePlayBonusCreditTableMockProps = {
  onSubmitDeletePlayFreeCredit: () => console.log('delete'),
  data: [
    {
      id: '1',
      transactionDate: new Date(),
      amountBonus: 100,
      status: PaymentStatusCode.COMPLETE,
      type: TransactionType.CREDIT,
    },
    {
      id: '2',
      transactionDate: new Date(),
      amountBonus: 100,
      status: PaymentStatusCode.CANCELLED,
      type: TransactionType.DEBIT,
    },
  ],
  totalBonusCreditBalance: 0,
};
