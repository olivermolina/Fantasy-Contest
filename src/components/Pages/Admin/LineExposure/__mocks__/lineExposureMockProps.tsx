import { League } from '@prisma/client';

export const lineExposureMockProps = {
  data: [
    {
      id: '1',
      name: 'LeBron James',
      category: 'Points',
      league: League.NBA,
      status: 'Scheduled',
      matchup: 'LAL @ DEN',
      total: 35,
      gamedate: '2023-01-23',
      gametime: '03:00:00 PM',
      underOdds: 100,
      overOdds: -120,
      overExposure: 4,
      underExposure: 2,
      betLegs: [
        {
          id: 'ticket-id',
          betId: 'bet-id',
          entryDate: '2023-01-22',
          payout: 100,
          stake: 10,
          type: 'OVER_ODDS',
          user: {
            id: 'user-1',
            firstName: 'John',
            lastName: 'Doe',
            username: 'jdoe',
          },
          status: 'WIN',
        },
      ],
    },
  ],
  setDateRange: () => console.log('setDateRange'),
  dateRange: null,
  selectedLeague: League.NBA,
  setSelectedLeague: () => console.log('setSelectedLeague'),
};
