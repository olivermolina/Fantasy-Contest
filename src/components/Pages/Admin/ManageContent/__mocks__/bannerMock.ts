import { AppSettingName } from '@prisma/client';

export const bannerMock = {
  id: '1',
  text: 'test',
  priority: 1,
  appSettingId: '1',
  appSetting: {
    id: '1',
    name: AppSettingName.BONUS_CREDIT_FREE_ENTRY_EQUIVALENT,
    value: '1',
  },
};
