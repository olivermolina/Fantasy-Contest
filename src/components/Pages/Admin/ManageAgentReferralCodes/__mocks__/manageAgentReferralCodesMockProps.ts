import { usersMock } from '~/components/Admin/Management/__mocks__/usersMocks';

export const manageAgentReferralCodesMockProps = {
  onSubmit: () => alert('test'),
  users: usersMock,
  isLoading: false,
  deleteReferralCode: () => alert('delete'),
};
