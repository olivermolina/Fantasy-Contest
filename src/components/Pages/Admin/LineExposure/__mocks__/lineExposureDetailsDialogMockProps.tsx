import { lineExposureMockProps } from './lineExposureMockProps';

export const lineExposureDetailsDialogMockProps = {
  row: lineExposureMockProps.data[0],
  open: false,
  handleClose: () => console.log(close),
};
