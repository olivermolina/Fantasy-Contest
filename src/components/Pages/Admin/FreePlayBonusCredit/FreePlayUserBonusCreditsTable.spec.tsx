import { render } from '@testing-library/react';
import FreePlayUserBonusCreditsTable from './FreePlayUserBonusCreditsTable';
import { freePlayBonusCreditTableMockProps } from './__mocks__/freePlayBonusCreditTableMockProps';

describe('FreePlayUserBonusCreditsTable', () => {
  it('should render correctly', () => {
    const { container } = render(
      <FreePlayUserBonusCreditsTable {...freePlayBonusCreditTableMockProps} />,
    );
    expect(container).toMatchSnapshot();
  });
});
