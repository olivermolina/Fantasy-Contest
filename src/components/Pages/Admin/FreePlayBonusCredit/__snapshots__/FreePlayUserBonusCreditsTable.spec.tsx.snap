// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FreePlayUserBonusCreditsTable should render correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-2"
  >
    <span
      class="text-lg font-bold"
    >
      Bonus Credit Transactions
    </span>
    <div
      style="height: 300px; width: 100%;"
    >
      <div
        aria-colcount="6"
        aria-multiselectable="false"
        aria-rowcount="3"
        class="MuiDataGrid-root MuiDataGrid-root--densityStandard MuiDataGrid-withBorderColor css-1lymaxv-MuiDataGrid-root"
        role="grid"
      >
        <div
          class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
        >
          <div
            class="MuiDataGrid-columnHeaders MuiDataGrid-withBorderColor css-1iyq7zh-MuiDataGrid-columnHeaders"
            style="min-height: 56px; max-height: 56px; line-height: 56px;"
          >
            <div
              class="MuiDataGrid-columnHeadersInner css-gl260s-MuiDataGrid-columnHeadersInner"
              role="rowgroup"
              style="transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-rowindex="1"
                class="css-yrdy0g-MuiDataGrid-columnHeaderRow"
                role="row"
              >
                <div
                  aria-colindex="1"
                  aria-label="Transaction ID"
                  aria-sort="none"
                  class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor"
                  data-field="id"
                  role="columnheader"
                  style="height: 56px; width: 0px; min-width: 0; max-width: 0;"
                  tabindex="0"
                >
                  <div
                    class="MuiDataGrid-columnHeaderDraggableContainer"
                    draggable="false"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainer"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainerContent"
                      >
                        <div
                          aria-label=""
                          class="MuiDataGrid-columnHeaderTitle css-t89xny-MuiDataGrid-columnHeaderTitle"
                          data-mui-internal-clone-element="true"
                        >
                          Transaction ID
                        </div>
                      </div>
                      <div
                        class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                      >
                        <button
                          aria-label="Sort"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                          tabindex="-1"
                          title="Sort"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                            data-testid="ArrowUpwardIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-menuIcon"
                    >
                      <button
                        aria-expanded="false"
                        aria-haspopup="menu"
                        aria-label="Menu"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        data-mui-internal-clone-element="true"
                        id=":r2:"
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                          data-testid="TripleDotsVerticalIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                    style="min-height: 56px; opacity: 1;"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                      data-testid="SeparatorIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 19V5h2v14z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  aria-colindex="2"
                  aria-label="Transaction Date"
                  aria-sort="none"
                  class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor"
                  data-field="transactionDate"
                  role="columnheader"
                  style="height: 56px; width: 0px; min-width: 0; max-width: 0;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-columnHeaderDraggableContainer"
                    draggable="false"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainer"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainerContent"
                      >
                        <div
                          aria-label=""
                          class="MuiDataGrid-columnHeaderTitle css-t89xny-MuiDataGrid-columnHeaderTitle"
                          data-mui-internal-clone-element="true"
                        >
                          Transaction Date
                        </div>
                      </div>
                      <div
                        class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                      >
                        <button
                          aria-label="Sort"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                          tabindex="-1"
                          title="Sort"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                            data-testid="ArrowUpwardIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-menuIcon"
                    >
                      <button
                        aria-expanded="false"
                        aria-haspopup="menu"
                        aria-label="Menu"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        data-mui-internal-clone-element="true"
                        id=":r8:"
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                          data-testid="TripleDotsVerticalIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                    style="min-height: 56px; opacity: 1;"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                      data-testid="SeparatorIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 19V5h2v14z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  aria-colindex="3"
                  aria-label="Type"
                  aria-sort="none"
                  class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor"
                  data-field="type"
                  role="columnheader"
                  style="height: 56px; width: 0px; min-width: 0; max-width: 0;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-columnHeaderDraggableContainer"
                    draggable="false"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainer"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainerContent"
                      >
                        <div
                          aria-label=""
                          class="MuiDataGrid-columnHeaderTitle css-t89xny-MuiDataGrid-columnHeaderTitle"
                          data-mui-internal-clone-element="true"
                        >
                          Type
                        </div>
                      </div>
                      <div
                        class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                      >
                        <button
                          aria-label="Sort"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                          tabindex="-1"
                          title="Sort"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                            data-testid="ArrowUpwardIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-menuIcon"
                    >
                      <button
                        aria-expanded="false"
                        aria-haspopup="menu"
                        aria-label="Menu"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        data-mui-internal-clone-element="true"
                        id=":re:"
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                          data-testid="TripleDotsVerticalIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                    style="min-height: 56px; opacity: 1;"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                      data-testid="SeparatorIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 19V5h2v14z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiDataGrid-virtualScroller css-axafay-MuiDataGrid-virtualScroller"
            style="overflow-x: hidden;"
          >
            <div
              class="MuiDataGrid-virtualScrollerContent MuiDataGrid-virtualScrollerContent--overflowed css-1kwdphh-MuiDataGrid-virtualScrollerContent"
              style="width: auto; height: 104px; min-height: auto;"
            >
              <div
                class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
                style="transform: translate3d(0px, 0px, 0px);"
              >
                <div
                  aria-rowindex="2"
                  aria-selected="false"
                  class="MuiDataGrid-row"
                  data-id="1"
                  data-rowindex="0"
                  role="row"
                  style="max-height: 52px; min-height: 52px;"
                >
                  <div
                    aria-colindex="1"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                    data-colindex="0"
                    data-field="id"
                    role="cell"
                    style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                      title="1"
                    >
                      1
                    </div>
                  </div>
                  <div
                    aria-colindex="2"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                    data-colindex="1"
                    data-field="transactionDate"
                    role="cell"
                    style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                      title="1/1/2021, 12:00:00 AM"
                    >
                      1/1/2021, 12:00:00 AM
                    </div>
                  </div>
                  <div
                    aria-colindex="3"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                    data-colindex="2"
                    data-field="type"
                    role="cell"
                    style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                      title="CREDIT"
                    >
                      CREDIT
                    </div>
                  </div>
                </div>
                <div
                  aria-rowindex="3"
                  aria-selected="false"
                  class="MuiDataGrid-row MuiDataGrid-row--lastVisible"
                  data-id="2"
                  data-rowindex="1"
                  role="row"
                  style="max-height: 52px; min-height: 52px;"
                >
                  <div
                    aria-colindex="1"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                    data-colindex="0"
                    data-field="id"
                    role="cell"
                    style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                      title="2"
                    >
                      2
                    </div>
                  </div>
                  <div
                    aria-colindex="2"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                    data-colindex="1"
                    data-field="transactionDate"
                    role="cell"
                    style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                      title="1/1/2021, 12:00:00 AM"
                    >
                      1/1/2021, 12:00:00 AM
                    </div>
                  </div>
                  <div
                    aria-colindex="3"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft MuiDataGrid-withBorderColor"
                    data-colindex="2"
                    data-field="type"
                    role="cell"
                    style="min-width: 0; max-width: 0; min-height: 52px; max-height: 52px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                      title="DEBIT"
                    >
                      DEBIT
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex p-2 text-lg gap-2 divide-x-2 items-center"
        >
          <p
            class="p-2"
          >
            Total Bonus Credits:
             
            <span
              class="font-semibold"
            >
              100.00
            </span>
          </p>
          <p
            class="p-2"
          >
            Total Bonus Credit Balance:
             
            <span
              class="font-semibold"
            >
              0.00
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
